packages/socket/patches/socket.io-parser+4.0.4.patch

```diff
@@ -0,0 +1,162 @@
+diff --git a/node_modules/socket.io-parser/dist/binary.js b/node_modules/socket.io-parser/dist/binary.js
+index a908023..99e0215 100644
+--- a/node_modules/socket.io-parser/dist/binary.js
++++ b/node_modules/socket.io-parser/dist/binary.js
+@@ -13,12 +13,12 @@ function deconstructPacket(packet) {
+     const buffers = [];
+     const packetData = packet.data;
+     const pack = packet;
+-    pack.data = _deconstructPacket(packetData, buffers);
++    pack.data = _deconstructPacket(packetData, buffers, [], new WeakMap());
+     pack.attachments = buffers.length; // number of binary 'attachments'
+     return { packet: pack, buffers: buffers };
+ }
+ exports.deconstructPacket = deconstructPacket;
+-function _deconstructPacket(data, buffers) {
++function _deconstructPacket(data, buffers, known, retvals) {
+     if (!data)
+         return data;
+     if (is_binary_1.isBinary(data)) {
+@@ -26,18 +26,27 @@ function _deconstructPacket(data, buffers) {
+         buffers.push(data);
+         return placeholder;
+     }
+-    else if (Array.isArray(data)) {
++    else if (retvals.has(data)) {
++        return retvals.get(data)
++    }
```

Comment:
this allows us to get a handle on the `newData` objects - socket.io takes care not to modify the incoming packet, so to keep these references correct in a circular object scenario, we must keep a map of data <=> newdata